interfaces/cython/cantera/thermo.pyx

```diff
@@ -1650,6 +1656,47 @@ cdef class ThermoPhase(_SolutionBase):
             cdef CxxUnits units = self.thermo.standardConcentrationUnits()
             return Units.copy(units)
 
+    # methods for plasma
+    def set_electron_energy_distribution(self, levels, distrb):
+        """ Set electron energy distribution. When this method is used, electron
+        temperature is calculated from the distribution.
+        :param levels:
+            vector of electron energy levels [eV]
+        :param distrb:
+            vector of distribution
+        """
+        cdef vector[double] cxxdata_levels
+        cdef vector[double] cxxdata_distrb
+        for value in levels:
+            cxxdata_levels.push_back(value)
+        for value in distrb:
+            cxxdata_distrb.push_back(value)
+        (<CxxPlasmaPhase*>self.thermo).setElectronEnergyDistribution(cxxdata_levels, cxxdata_distrb)
```

Comment:
You need to add a check that `self.thermo` has the correct type (`"Plasma"`) and raise an exception if it does not. Otherwise, calling this function for a non-plasma object will have unexpected behavior.

Same for the other `Plasma`-specific methods/properties.

Reply:
Are TypeError and its messages good?